.github/workflows/s3-deploy.yml

```diff
@@ -0,0 +1,54 @@
+name: Deploy to S3
```

Comment:
I don't know if having this as a separate workflow will work. I think as a separate workflow, it will be run separately so the `dist` folder you wanted to uploader won't exist. May have to either rebuild the library in steps, then upload. Or move these steps into the previous workflow and guard them with conditions.